src/lib.rs

```diff
@@ -25,15 +26,22 @@ use ark_std::{
     iter::FromIterator,
     rc::Rc,
     string::{String, ToString},
+    vec,
     vec::Vec,
 };
+use hashbrown::{HashMap, HashSet};
 
 /// Data structures used by a polynomial commitment scheme.
 pub mod data_structures;
 pub use data_structures::*;
 
+/// Constraints for recursion.
+mod pc_constraints;
+pub use pc_constraints::*;
```

Comment:
Let's rename the file to just `constraints.rs`, and also put it behind a feature flag of `r1cs`.

```suggestion
/// R1CS constraints for polynomial constraints.
#[cfg(feature = "r1cs")]
mod constraints;
#[cfg(feature = "r1cs")]
pub use constraints::*;
```